src/devices/vdml_imu.c

```diff
@@ -156,3 +191,106 @@ imu_status_e_t imu_get_status(uint8_t port) {
 	rtn = vexDeviceImuStatusGet(device->device_info);
 	return_port(port - 1, rtn);
 }
+
+//Reset Functions:
+int32_t imu_tare(uint8_t port){
+    if (!claim_port_try(port - 1, E_DEVICE_IMU)) {
+		return PROS_ERR;
+	}
+	v5_smart_device_s_t* device = registry_get_device(port - 1);
+	euler_s_t euler_values;
+	vexDeviceImuAttitudeGet(device->device_info, (V5_DeviceImuAttitude*)&euler_values);
+	imu_data_s_t* data = (imu_data_s_t*)device->pad;
+	data->rotation_offset = -vexDeviceImuHeadingGet(device->device_info);
+	data->heading_offset = -vexDeviceImuDegreesGet(device->device_info);
+	data->pitch_offset = -euler_values.pitch;
+	data->roll_offset = -euler_values.roll;
+	data->yaw_offset = -euler_values.yaw;
+	return_port(port - 1, 1);
```

Comment:
```suggestion
	return_port(port - 1, PROS_SUCCESS);
```

Reply:
For all of these that return 1

Reply:
Note for future reference: PROS_SUCCESS does not exist but we should add it in PROS 4 for readability's sake.